src/python/pants/option/global_options.py

```diff
@@ -726,6 +726,14 @@ def register_bootstrap_options(cls, register):
         )
 
         # Watchman options.
+        register(
+            "--watchman-enable",
```

Comment:
Question: would there ever be a reason to use neither watchman nor libnotify or to use both?
If the answer is no, it would probably better to keep these 2 options consolidated into one for better ergonomicsb

Reply:
We have used both watchman and notify at the same time to test out notify feature partity, or better compared to watchman. This is something we may want to do as we enable this feature downstream.

Reply:
Thanks for clarifying. In that case, it makes sense to expose both with their own flags. Marking this as resoved.